mvc/controllers/areas.rst

```diff
@@ -1,8 +1,132 @@
-.. include:: /../common/stub-topic.txt
+Areas
+=====
+By `Tom Archer`
 
-|stub-icon| Areas
-===========================================
+MVC uses a folder hierarchy pattern that defines the relationship between your controllers, views, and actions. You can see this with the default wizard-generated ASP.NET MVC app where you start with a ``Controllers`` folder containing several controllers (``Account``, ``Home``, and ``Manage``). If you open the ``HomeController.cs`` file, you'll note that the ``HomeController`` controller has three actions - ``Index``, ``About``, and ``Contact`` - that return a view object without qualifying which view to return. (A fourth action - ``Error`` - is present, but it explicitly names the location of the view to return.) So, how does MVC know which view to return when its location is not explicitly stated?
 
-.. include:: /../common/stub-notice.txt
+MVC finds the associated view for a controller action by searching in two folders: ``Views/[controller][action].cshtml`` and ``Views/Shared/[action].cshtml``. Therefore, when the ``HomeController.About`` method attempts to create a view without specifying the location of the view, the MVC routing engine will look for ``Views/Home/About.cshtml`` first. If that's not found, the routing engine will look for ``Views/Shared/About.cshtml``. If neither are found, an exception is thrown.
 
-.. _issue: https://github.com/aspnet/Docs/issues/121
+This pattern works well for many Web apps. However, for larger apps, this folder hierarchy can become difficult to manage as you have controllers in one parent folder, and views in a separate parent folder. This is where `Areas` come in. Areas enable you to partition your project such that controllers and their views can be stored under a parent folder that defines their logical grouping.
+
+Let's take a look at an example to illustrate how Areas are created and used. In this example, we'll take a simple store app (MyStore) that has a several logical groupings of controllers and actions. These groupings are ``Services``, ``Products``, and ``Deals``. The following steps will walk you through creating an area for each grouping, as well as the controllers and views for each.
+
+#. In Solution Explorer, right-click the project name and create a new folder named ``Areas``.
```

Comment:
+1. Remove the point and click instructions and instead focus on documenting the runtime behavior.